nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SNMPUtils.java

```diff
@@ -0,0 +1,268 @@
+import org.snmp4j.security.PrivAES256;
+import org.snmp4j.security.PrivDES;
+import org.snmp4j.security.SecurityLevel;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.VariableBinding;
+import org.snmp4j.util.TreeEvent;
+
+/**
+ * Utility helper class that simplifies interactions with target SNMP API and NIFI API.
+ */
+abstract class SNMPUtils {
+
+    /** logger */
+    private final static Logger logger = LoggerFactory.getLogger(SNMPUtils.class);
+
+    /** OID Pattern */
+    public final static Pattern OID_PATTERN = Pattern.compile("[[0-9]+\\.]*");
+
+    /** delimiter for properties name */
+    public final static String SNMP_PROP_DELIMITER = "$";
+
+    /** prefix for SNMP properties in flow file */
+    public final static String SNMP_PROP_PREFIX = "snmp" + SNMP_PROP_DELIMITER;
+
+    /** list of properties name when performing simple get */
+    private final static List<String> propertyNames = Arrays.asList("snmp$errorIndex", "snmp$errorStatus", "snmp$errorStatusText",
+            "snmp$nonRepeaters", "snmp$requestID", "snmp$type", "snmp$variableBindings");
+
+    /** used to validate OID syntax */
+    public static final Validator SNMP_OID_VALIDATOR = new Validator() {
+        @Override
+        public ValidationResult validate(final String subject, final String input, final ValidationContext context) {
+            final ValidationResult.Builder builder = new ValidationResult.Builder();
+            builder.subject(subject).input(input);
+            if (context.isExpressionLanguageSupported(subject) && context.isExpressionLanguagePresent(input)) {
+                return builder.valid(true).explanation("Contains Expression Language").build();
+            }
+            try {
+                if (OID_PATTERN.matcher(input).matches()) {
+                    builder.valid(true);
+                } else {
+                    builder.valid(false).explanation(input + "is not a valid OID");
+                }
+            } catch (final IllegalArgumentException e) {
+                builder.valid(false).explanation(e.getMessage());
+            }
+            return builder.build();
+        }
+    };
+
+    /**
+     * Updates {@link FlowFile} with attributes representing PDU properties
+     * @param response PDU retried from SNMP Agent
+     * @param flowFile instance of target {@link FlowFile}
+     * @param processSession instance of {@link ProcessSession}
+     * @return updated {@link FlowFile}
+     */
+    public static FlowFile updateFlowFileAttributesWithPduProperties(PDU response, FlowFile flowFile, ProcessSession processSession) {
+        if (response != null) {
+            try {
+                Method[] methods = PDU.class.getDeclaredMethods();
+                Map<String, String> attributes = new HashMap<String, String>();
+                for (Method method : methods) {
+                    if (Modifier.isPublic(method.getModifiers()) && (method.getParameterCount() == 0) && method.getName().startsWith("get")) {
```

Comment:
Pierre, I noticed this when I was researching for your OnStopped issue. I believe this method 'getParameterCount' was introduced in Java 8 while we still have to be compliant with Java 7. So consider refactoring.